binderhub/static/js/components/LinkGenerator.jsx

```diff
@@ -0,0 +1,347 @@
+    }
+  }
+
+  return (
+    <form className={`d-flex flex-column gap-3 ${className}`}>
+      <h4>Build and launch a repository</h4>
+      <fieldset>
+        <legend className="form-label">{selectedProvider.repo.label}</legend>
+        <div className="input-group">
+          <ProviderSelector
+            providers={providers}
+            selectedProvider={selectedProvider}
+            setSelectedProvider={setSelectedProvider}
+          />
+          <input
+            className="form-control border border-2 border-start-0"
+            type="text"
+            name="repository"
+            placeholder={selectedProvider.repo.placeholder}
+            disabled={isLaunching}
+            aria-label="Enter repository URL"
+            onChange={(e) => {
+              let repo = e.target.value;
+              if (selectedProvider.detect && selectedProvider.detect.regex) {
+                // repo value *must* be detected by this regex, or it is not valid yet
+                const re = new RegExp(selectedProvider.detect.regex);
+                const results = re.exec(repo);
+                if (results !== null && results.groups && results.groups.repo) {
+                  setRepo(results.groups.repo);
+                }
+              } else {
+                setRepo(e.target.value);
+              }
+            }}
+          />
+        </div>
+      </fieldset>
+
+      <div className="row align-items-end">
+        <div className="col-5">
+          <label htmlFor="ref" className="form-label">
+            Git ref (branch, tag, or commit)
+          </label>
+          <div className="input-group">
+            <input
+              className="form-control border border-2"
+              type="text"
+              id="ref"
+              name="ref"
+              disabled={!selectedProvider.ref.enabled || isLaunching}
+              placeholder={
+                (selectedProvider.ref.enabled &&
+                  selectedProvider.ref.default) ||
+                ""
+              }
+              onChange={(e) => {
+                setReference(e.target.value);
+              }}
+            />
+          </div>
+        </div>
+        <div className="col-5">
+          <UrlSelector setUrlPath={setUrlPath} />
+        </div>
+        <div className="col-2">
+          <button
+            className="btn btn-primary col-2 w-100"
+            disabled={isLaunching}
+            onClick={() => setIsLaunching(true)}
+          >
+            {isLaunching ? "launching..." : "launch"}
+          </button>
+        </div>
+      </div>
+
+      <div>
+        <div className="input-group">
+          <div
+            className="form-control font-monospace border border-2 border-end-0"
+            data-testid="launch-url"
+          >
+            {launchUrl ||
+              "Fill in the fields to see a URL for sharing your Binder."}
+          </div>
+          <button
+            className="btn border border-2 border-start-0"
```

Comment:
```suggestion
            className="btn btn-outline-secondary border border-2 border-start-0"
```
The button is currently missing a hover effect.